iib/web/models.py

```diff
@@ -365,8 +409,52 @@ def from_json(cls, kwargs):
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
+    @classmethod
+    def from_add_json(cls, kwargs):
+        """ Handles JSON requests for Add API endpoint """
+        request_kwargs = deepcopy(kwargs)
+
+        bundles = request_kwargs.pop('bundles', [])
+        if (
+            not isinstance(bundles, list)
+            or len(bundles) == 0
+            or any(not item or not isinstance(item, str) for item in bundles)
+        ):
+            raise ValidationError(f'"bundles" should be a non-empty array of strings')
+
+        request_kwargs['bundles'] = [
```

Comment:
If you don't `pop` the `bundles` key, you can perform this after the `cls._from_json`, which means the database isn't touched until all validation has occurred which is nice.

Reply:
Good idea!